src/services/TestResultsService.ts

```diff
@@ -278,11 +278,29 @@ export class TestResultsService {
                     payload.testTypes[index] = testType;
                   }
                 } else if (payload.vehicleType === VEHICLE_TYPES.HGV || payload.vehicleType === VEHICLE_TYPES.TRL) {
+                  // Checks for testType = First test or First test Retest AND if vehicle doesn't currently have an existing expiry date, i.e set to default
+                  if (this.isFirstTestRetestTestType(testType) && dateFns.isEqual(mostRecentExpiryDateOnAllTestTypesByVin, new Date(1970, 1, 1))) {
+                    // Applying CVSB-8658 logic changes if vehicle doesn't currently have an existing expiry date
+                    const regOrFirstUseDate = payload.vehicleType === VEHICLE_TYPES.HGV ? payload.regnDate : payload.firstUseDate;
+                    const anvDateForCompare = regOrFirstUseDate ? dateFns.addYears(dateFns.lastDayOfMonth(regOrFirstUseDate), 1).toISOString() : undefined;
+                    console.log(`Test Date: ${new Date()}, Anv Date: ${anvDateForCompare}`);
+                    // If anniversaryDate is not populated in tech-records OR test date is 2 months or more before the Registration/First Use Anniversary for HGV/TRL
+                    if(!anvDateForCompare || dateFns.isBefore(new  Date(), dateFns.subMonths(anvDateForCompare, 2))){
+                      testType.testExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1).toISOString();
+                      console.log(`Setting expiryDate: ${testType.testExpiryDate}`);
+                    }else {
+                      // less than 2 months then set expiryDate 1 year after the Registration/First Use Anniversary date
+                      testType.testExpiryDate = dateFns.addYears(anvDateForCompare, 1).toISOString();
+                      console.log(`Setting expiryDate 1yr after AnvDate: ${testType.testExpiryDate}`);
```

Comment:
Will remove the console logs once testing is completed using postman (waiting for 8703 to be done).